testing/sdk_auth.py

```diff
@@ -0,0 +1,376 @@
+        raise RuntimeError("Unable to get the master node's public IP address: {err}".format(err=repr(raw_response)))
+
+    response = json.loads(raw_response.text)
+    if "PUBLIC_IPV4" not in response:
+        raise KeyError("Cluster metadata does not include master's public ip: {response}".format(
+            response=repr(raw_response)))
+
+    public_ip = response["PUBLIC_IPV4"]
+    log.info("Master public ip is {public_ip}".format(public_ip=public_ip))
+    return public_ip
+
+
+def _create_temp_working_dir() -> str:
+    """
+    Creates a temporary working directory to enable setup of the Kerberos environment.
+    :return (str): The path of the working directory.
+    """
+    dir_name = "/tmp/{kerberos}_{current_time}".format(kerberos=KERBEROS_APP_ID, current_time=str(time.time()))
+    log.info("Creating temp working directory {}".format(dir_name))
+    try:
+        os.makedirs(dir_name)
+    except OSError as e:
+        raise OSError("Can't create temp working directory: {}".format(repr(e)))
+
+    return dir_name
+
+
+def _copy_file_to_localhost(self):
+    """
+    Copies the keytab that was generated inside the container running the KDC server to the localhost
+    so it can be uploaded to the secret store later. This must be done in multiple steps as we have to jump through
+    hoops.
+
+    The keytab will end up in path: <temp_working_dir>/<keytab_file>
+    """
+    log.info("Copying {} to the temp working directory".format(self.keytab_file_name))
+
+    # 1. copy from within container to private agent
+    copy_cmd = "sudo cp /var/lib/mesos/slave/slaves/{host_id}/frameworks/{framework_id}/executors/{task_id}/runs/latest/{keytab_file} /home/{linux_user}/{keytab_file}".format(
+        host_id=self.kdc_host_id,
+        framework_id=self.framework_id,
+        task_id=self.task_id,
+        keytab_file=self.keytab_file_name,
+        linux_user=LINUX_USER
+    )
+
+    try:
+        shakedown.run_command_on_agent(self.kdc_host_name, copy_cmd)
+    except dcos.errors.DCOSException as e:
+        raise RuntimeError("Failed to copy keytab file from container to agent: {}".format(repr(e)))
+
+    # 2. Change file ownership so it can be scp'ed to leader
+    chown_cmd = "sudo chown {linux_user} {keytab_file}".format(linux_user=LINUX_USER, keytab_file=self.keytab_file_name)
+
+    try:
+        shakedown.run_command_on_agent(self.kdc_host_name, chown_cmd)
+    except dcos.errors.DCOSException as e:
+        raise RuntimeError("Failed to chown keytab file: {}".format(repr(e)))
+
+    # 3. Copy from private agent to leader
+    scp_cmd = "scp {linux_user}@{kdc_host_name}:/home/{linux_user}/{keytab_file} /home/{linux_user}/{keytab_file}".format(
+        linux_user=LINUX_USER,
+        kdc_host_name=self.kdc_host_name,
+        keytab_file=self.keytab_file_name
+    )
+    cmd = "eval $(ssh-agent -s) && ssh-add /ssh/key && dcos node ssh --master-proxy --leader --option StrictHostKeyChecking=no '{}'".format(scp_cmd)
```

Comment:
Should the key be configurable? This will, for example, not work in our local `test.sh` runs, ad we there mount the key file at a different path.

Reply:
So it seems like the scripts use `/ssh/key` as the mount directory, in both local runs and CI. I mean, this worked for me locally, so I may be missing something you're referring to?

Reply:
I checked again, and I stand corrected. I think there was some other reason that this was not working for me.